### PR TITLE
[stable/mongodb-replicaset] Dont initiate replicaset during bootstrapping

### PR DESCRIPTION
**Pull request by ghost**
#### What this PR does / why we need it:

@steven-sheehy @unguiculus 

This skips initialization of the replicaset during bootstraping if set to true. This is helpful during a migration phase. You can now add the created nodes to an already existing MongoDB replicaset and still install and mange the cluster with the helm chart.

You just end up with the defined number of new empty nodes and can implement your own migration path with it.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/mongodb-replicaset/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: mongodb-replicaset
 home: https://github.com/mongodb/mongo
-version: 3.10.2
+version: 3.10.3
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/Chart.yaml]: `3.11.0` for new features

File: stable/mongodb-replicaset/README.md [hunk 2]
```diff
@@ -36,6 +36,7 @@ The following table lists the configurable parameters of the mongodb chart and t
 | ----------------------------------- | ------------------------------------------------------------------------- | --------------------------------------------------- |
 | `replicas`                          | Number of replicas in the replica set                                     | `3`                                                 |
 | `replicaSetName`                    | The name of the replica set                                               | `rs0`                                               |
+| `doNotInit`                    | If `true` skip replica set initialization during bootstrapping                                              | `false`
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/README.md]: `skipInitialization` and change other related variables

File: stable/mongodb-replicaset/init/on-start.sh [hunk 3]
```diff
@@ -151,67 +151,71 @@ fi
...
-    log "This replica is already PRIMARY"
-elif [[ -n "${primary}" ]]; then
-    if [[ $(mongo admin --host "${primary}" "${admin_creds[@]}" "${ssl_args[@]}" --quiet --eval "rs.conf().members.findIndex(m => m.host == '${service_name}:${port}')") == "-1" ]]; then
-      log "Adding myself (${service_name}) to replica set..."
-      if (mongo admin --host "${primary}" "${admin_creds[@]}" "${ssl_args[@]}" --eval "rs.add('${service_name}')" | grep 'Quorum check failed'); then
-          log 'Quorum check failed, unable to join replicaset. Exiting prematurely.'
-          exit 1
+if [[ "${DO_NOT_INIT}" == "true" ]]; then
+    log "Skipping initialization"
+else
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/init/on-start.sh]: Don't add an else, just fail fast by logging and exit 0.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
Not sure I see the need for this when we have the [initMongodStandalone](https://github.com/helm/charts/tree/master/stable/mongodb-replicaset#indexes-and-maintenance) functionality to run arbitrary scripts before the replicaset is started. What use case does this solve that that feature does not handle?

**Comment by ofcourseican** [comment]
> Not sure I see the need for this when we have the [initMongodStandalone](https://github.com/helm/charts/tree/master/stable/mongodb-replicaset#indexes-and-maintenance) functionality to run arbitrary scripts before the replicaset is started. What use case does this solve that that feature does not handle?

From my understanding the`initMongodStandalone`flag does not skip the whole replicaset creation process. It just allows to run JS on each of the instances during bootstraping time.

What I want is a completely skipped replicaset creation process. We use those pods to integrate them into an existing mongodb replicaset (using the Kubernetes external services feature: https://kubernetes.io/docs/concepts/services-networking/service/#externalname). The goal is to migrate fully from VMs into Kubernetes. By doing it this way we can keep the Helm chart deployment after we are done with he migration, we just remove the external replica services.

Might be a bit of a special use-case, but I think especially during migrations a feature like this is very useful. It leaves you with the option to keep using the helm chart to install the new nodes and keep using this deployment after the migration is done.

**Comment by steven-sheehy** [comment]
@ofcourseican Please fix conflict. Also, please add a section to README about why this option is useful and how to use it.

File: stable/mongodb-replicaset/Chart.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: mongodb-replicaset
 home: https://github.com/mongodb/mongo
-version: 3.10.3
+version: 3.11.1
```
Comment (steven-sheehy) [inline stable/mongodb-replicaset/Chart.yaml]: `3.11.0` and it should be good to go.

**Comment by ofcourseican** [comment]
@steven-sheehy Could you have another look at this, please?

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Review by steven-sheehy — APPROVED** [review]
/lgtm
Thanks!

**Pull request merged**